src/Apps/Order/Components/ExpressCheckout/Order2ExpressCheckoutUI.tsx

```diff
@@ -0,0 +1,729 @@
+            },
+          },
+        })
+
+      validateAndExtractOrderResponse(
+        updateOrderShippingAddressResult.updateOrderShippingAddress
+          ?.orderOrError,
+      )
+
+      // Trigger form validation and wallet collection
+      const { error: submitError } = await elements.submit()
+      if (submitError) {
+        logger.error(submitError)
+        return
+      }
+
+      // Create the ConfirmationToken using the details collected by the Payment Element
+      const { error, confirmationToken } = await stripe.createConfirmationToken(
+        {
+          elements,
+          params:
+            shippingRate?.id === "PICKUP"
+              ? {
+                  shipping: {
+                    address: {
+                      line1: null,
+                      line2: null,
+                      city: null,
+                      postal_code: null,
+                      state: null,
+                      country: null,
+                    },
+                    name: null,
+                  },
+                }
+              : undefined,
+        },
+      )
+
+      if (error) {
+        // This point is only reached if there's an immediate error when
+        // creating the ConfirmationToken. Show the error to customer (for example, payment details incomplete)
+        logger.error(error)
+        return
+      }
+
+      setShowSpinner?.(true)
+
+      const submitOrderResult = await submitOrderMutation.submitMutation({
+        variables: {
+          input: {
+            id: orderData.internalID,
+            confirmationToken: confirmationToken.id,
+          },
+        },
+      })
+
+      validateAndExtractOrderResponse(
+        submitOrderResult.submitOrder?.orderOrError,
+      )
+
+      // Redirect to status page after successful order submission
+      window.location.reload()
+    } catch (error) {
+      logger.error("Error confirming payment", error)
+      errorRef.current = error.code || "unknown_error"
+
+      orderTracking.errorMessageViewed({
+        error_code: errorRef.current,
+        title: "Payment failed",
+        message:
+          "Something went wrong. Please try again or contact orders@artsy.net",
+        flow: "Express checkout",
+      })
+
+      sessionStorage.setItem(
+        "expressCheckoutError",
+        JSON.stringify({
+          title: "Payment failed",
+        }),
+      )
+
+      resetOrder()
+    }
+  }
+
+  const handleReady = e => {
+    try {
+      const paymentMethods = e.availablePaymentMethods
+
+      if (paymentMethods) {
+        setVisible(true)
+
+        orderTracking.expressCheckoutViewed({
+          order: orderData,
+          walletType: getAvailablePaymentMethods(paymentMethods),
+        })
+      }
+    } catch (error) {
+      logger.error("Error handling ready event", error)
+    } finally {
+      setExpressCheckoutLoaded(e.availablePaymentMethods)
+    }
+  }
```

Comment:
I think that we want to use `getAvailablePaymentMethods` on this to extract the keys from the object which are true. See https://github.com/artsy/force/blob/aa2fd08ca305a59a6f689e24b080474886319acd/src/Apps/Order/Components/ExpressCheckout/ExpressCheckoutUI.tsx#L717-L723

The naming of this event from stripe and its corresponding type are misleading:
```ts
export type AvailablePaymentMethods = {
  amazonPay: boolean;
  applePay: boolean;
  googlePay: boolean;
  link: boolean;
  paypal: boolean;
};
```